### PR TITLE
Allow custom OpenSSL lib, only include openssl/x509v3.h when used.

### PR DESCRIPTION
**Pull request by mattparks**
Using fnmatch on Unix systems, PathMatchSpecA is the best WINAPI equivalent.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -175,11 +175,13 @@ if (USE_TLS AND USE_OPEN_SSL)
...
 
-  find_package(OpenSSL REQUIRED)
+  if(NOT OPENSSL_FOUND)
+    find_package(OpenSSL REQUIRED)
+  endif()
   add_definitions(${OPENSSL_DEFINITIONS})
   message(STATUS "OpenSSL: " ${OPENSSL_VERSION})
   include_directories(${OPENSSL_INCLUDE_DIR})
-  target_link_libraries(ixwebsocket ${OPENSSL_LIBRARIES})
+  target_link_libraries(ixwebsocket ${OPENSSL_LIBRARIES} $<$<PLATFORM_ID:Windows>:Shlwapi.lib>)
```
Comment (bsergean) [inline CMakeLists.txt]: Can you add this target_link_libraries inside an if (WIN32)

Comment (mattparks) [reply CMakeLists.txt]: I moved `shlwapi` into a WIN32 block.

**Comment by bsergean** [comment]
Merged, thanks for the help !

What would be good to do now is 

1. bump the version number and add a small changelog note.
2. change the appveyor file to test the library when using openssl.

Have you tried to run the unittest ?

**Comment by mattparks** [comment]
I haven't tried to run the unittest, I made sure the entire project still built on Windows, iOS, and Android.

**Comment by bsergean** [comment]
All good. If you ever get a chance to investigate how to run the unittest on Windows that will be ideal.

I need to setup CI directly in that project. There's CI setup in my fork but it is a bit manual.

**Comment by bsergean** [comment]
Hey @mattparks fyi I implemented a Windows feature to setup the cacert automatically in f9d75c937408b80aa4f5ff0551582d016298eda4 if you want to give it a try.

**Pull request merged**